codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -180,6 +181,27 @@ DefaultHttp2FrameStream newStream() {
         return new DefaultHttp2FrameStream();
     }
 
+    /**
+     * Creates a new outbound/local stream.
+     */
+    DefaultHttp2FrameStream newStream(int streamId) throws Http2Exception {
+        assert ctx.executor().inEventLoop();
+        DefaultHttp2FrameStream stream = new DefaultHttp2FrameStream();
+        frameStreamToInitializeMap.put(streamId, stream);
+        try {
+            // We don't create the stream in a `halfClosed` state because if this is an initial
+            // HEADERS frame we don't want the connection state to signify that the HEADERS have
+            // been sent until after they have been encoded and placed in the outbound buffer.
+            // Therefore, we let the `LifeCycleManager` will take care of transitioning the state
+            // as appropriate.
+            connection().local().createStream(streamId, /*endOfStream*/ false);
```

Comment:
This unfortunately puts user in charge of maintaining id -> stream map (and removal on stream termination) to correlate frames - as this code is allowed to be called only once per stream. I was seeking to avoid that - codec maintains map already, users care about streamId only

Reply:
The removal should be done under the hood. Maybe we should add a method to `Http2FrameCodec` that allows to get the `Http2StreamFrame` for an id ?